From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Kleinschmidt <jkleinsc@electronjs.org>
Date: Wed, 26 Oct 2022 10:59:53 -0400
Subject: Revert "Use accessibility.pkey when setting page access."

This reverts commit 1c803026432e70740a319a51b41415694ebe36c9.

https://chromium-review.googlesource.com/c/chromium/src/+/3949281
breaks builds running in Docker.

This patch should be removed once
https://chromium-review.googlesource.com/c/chromium/src/+/3949284
is merged.

diff --git a/base/allocator/partition_allocator/page_allocator_internals_posix.h b/base/allocator/partition_allocator/page_allocator_internals_posix.h
index 6d2f55fa5f7984755517905adabb278ea835a295..455f06f57d3d92b2a24a6c8bf09b3deacfc088f3 100644
--- a/base/allocator/partition_allocator/page_allocator_internals_posix.h
+++ b/base/allocator/partition_allocator/page_allocator_internals_posix.h
@@ -201,14 +201,8 @@ bool TrySetSystemPagesAccessInternal(
     uintptr_t address,
     size_t length,
     PageAccessibilityConfiguration accessibility) {
-#if BUILDFLAG(ENABLE_PKEYS)
-  return 0 == PkeyMprotectIfEnabled(reinterpret_cast<void*>(address), length,
-                                    GetAccessFlags(accessibility),
-                                    accessibility.pkey);
-#else
   return 0 == PA_HANDLE_EINTR(mprotect(reinterpret_cast<void*>(address), length,
                                        GetAccessFlags(accessibility)));
-#endif
 }
 
 void SetSystemPagesAccessInternal(
@@ -216,14 +210,8 @@ void SetSystemPagesAccessInternal(
     size_t length,
     PageAccessibilityConfiguration accessibility) {
   int access_flags = GetAccessFlags(accessibility);
-#if BUILDFLAG(ENABLE_PKEYS)
-  int ret =
-      PkeyMprotectIfEnabled(reinterpret_cast<void*>(address), length,
-                            GetAccessFlags(accessibility), accessibility.pkey);
-#else
-  int ret = PA_HANDLE_EINTR(mprotect(reinterpret_cast<void*>(address), length,
+  const int ret = PA_HANDLE_EINTR(mprotect(reinterpret_cast<void*>(address), length,
                                      GetAccessFlags(accessibility)));
-#endif
 
   // On Linux, man mprotect(2) states that ENOMEM is returned when (1) internal
   // kernel data structures cannot be allocated, (2) the address range is
